.github/workflows/publish-snapshot-docker.yml

```diff
@@ -29,14 +29,22 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
       - name: Login to Docker Hub
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USER }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build Kyuubi Docker Image
-        run: docker build --tag apache/kyuubi:master-snapshot --file build/Dockerfile .
-      - name: Docker image
-        run: docker images
-      - name: Push Docker image
-        run: docker push apache/kyuubi:master-snapshot
+      - name: Build and Push Kyuubi Docker Image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
+          push: true
+          file: build/Dockerfile
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          tags: apache/kyuubi:master-snapshot
```

Comment:
Reorder configs in alphabetical order or in descending order of importance.

Reply:
Ok